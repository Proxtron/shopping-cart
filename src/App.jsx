import { Outlet } from 'react-router';
import Header from './layout/Header'
import { useState } from 'react';

function App() {
  const [numberInCart, setNumberInCart] = useState(0);

  function addToNumberInCart(amount) {
    setNumberInCart(numberInCart + amount);
  } 

  const context = {addToNumberInCart}
  return (
    <>
      <Header numberInCart={numberInCart}/>
      <main>
        <Outlet context={context}/>
      </main>
    </>
  )
}

export default App
