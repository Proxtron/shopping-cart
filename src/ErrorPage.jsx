import { useRouteError, Link } from "react-router";
import styles from "./ErrorPage.module.css";

const ErrorPage = () => {
  const error = useRouteError();

  // Determine error message and status
  const getErrorInfo = () => {
    if (error?.status === 404) {
      return {
        status: 404,
        title: "Page Not Found",
        message: "The page you're looking for doesn't exist.",
        suggestion: "Check the URL or go back to our homepage.",
      };
    } else if (error?.status >= 500) {
      return {
        status: error.status || 500,
        title: "Server Error",
        message: "Something went wrong on our end.",
        suggestion:
          "Please try again later or contact support if the problem persists.",
      };
    } else {
      return {
        status: error?.status || "Unknown",
        title: "Something Went Wrong",
        message: error?.message || "An unexpected error occurred.",
        suggestion:
          "Please try refreshing the page or go back to our homepage.",
      };
    }
  };

  const errorInfo = getErrorInfo();

  return (
    <div className={styles.errorPage}>
      <div className={styles.errorContainer}>
        <div className={styles.errorContent}>
          <h1 className={styles.errorStatus}>{errorInfo.status}</h1>
          <h2 className={styles.errorTitle}>{errorInfo.title}</h2>
          <p className={styles.errorMessage}>{errorInfo.message}</p>
          <p className={styles.errorSuggestion}>{errorInfo.suggestion}</p>

          <div className={styles.errorActions}>
            <Link to="/" className={styles.primaryButton}>
              Go to Homepage
            </Link>
            <button
              onClick={() => window.location.reload()}
              className={styles.secondaryButton}
            >
              Refresh Page
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default ErrorPage;
