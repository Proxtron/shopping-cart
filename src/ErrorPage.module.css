.errorPage {
  min-height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
  font-family: "Helvetica Neue", sans-serif;
}

.errorContainer {
  max-width: 600px;
  width: 100%;
  padding: 2rem;
}

.errorContent {
  text-align: center;
  background: #FEFEFE;
  padding: 3rem 2rem;
  border-radius: 8px;
}

.errorStatus {
  font-size: 6rem;
  font-weight: 700;
  margin: 0 0 1rem 0;
  line-height: 1;
}

.errorTitle {
  font-size: 2rem;
  font-weight: 500;
  color: #212529;
  margin: 0 0 1rem 0;
}

.errorMessage {
  font-size: 1.125rem;
  color: #6c757d;
  margin: 0 0 1rem 0;
  line-height: 1.5;
}

.errorSuggestion {
  font-size: 1rem;
  color: #6c757d;
  margin: 0 0 2rem 0;
  line-height: 1.5;
}

.errorActions {
  display: flex;
  gap: 1rem;
  justify-content: center;
  flex-wrap: wrap;
}

.primaryButton {
  display: inline-block;
  padding: 0.75rem 1.5rem;
  background-color: #007bff;
  color: white;
  text-decoration: none;
  border-radius: 4px;
  font-weight: 500;
  transition: background-color 0.2s;
}

.primaryButton:hover {
  background-color: #0056b3;
  color: white;
}

.secondaryButton {
  padding: 0.75rem 1.5rem;
  background-color: transparent;
  color: #6c757d;
  border: 2px solid #dee2e6;
  border-radius: 4px;
  font-weight: 500;
  cursor: pointer;
  transition: all 0.2s;
}

.secondaryButton:hover {
  background-color: #f8f9fa;
  border-color: #adb5bd;
  color: #495057;
}

/* Responsive design */
@media (max-width: 768px) {
  .errorContainer {
    padding: 1rem;
  }

  .errorContent {
    padding: 2rem 1.5rem;
  }

  .errorStatus {
    font-size: 4rem;
  }

  .errorTitle {
    font-size: 1.5rem;
  }

  .errorActions {
    flex-direction: column;
    align-items: center;
  }

  .primaryButton,
  .secondaryButton {
    width: 100%;
    max-width: 200px;
  }
}
