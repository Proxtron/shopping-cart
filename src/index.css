@font-face {
  font-family: "Helvetica Neue";
  src: url(./assets/Helvetica\ Neue/HelveticaNeue-Thin.otf) format("opentype");
  font-weight: 100;
  font-style: normal;
}

@font-face {
  font-family: "Helvetica Neue";
  src: url(./assets/Helvetica\ Neue/HelveticaNeue-Light.otf) format("opentype");
  font-weight: 300;
  font-style: normal;
}

@font-face {
  font-family: "Helvetica Neue";
  src: url(./assets/Helvetica\ Neue/HelveticaNeue-Medium.otf) format("opentype");
  font-weight: 500;
  font-style: normal;
}

@font-face {
  font-family: "Helvetica Neue";
  src: url(./assets/Helvetica\ Neue/HelveticaNeue-Bold.otf) format("opentype");
  font-weight: 700;
  font-style: normal;
}
@font-face {
  font-family: "Uniqlo Pro Light Regular";
  src: url("https://db.onlinewebfonts.com/t/13bf1d3e6c7119293f9c96e5678096e6.eot");
  src: url("https://db.onlinewebfonts.com/t/13bf1d3e6c7119293f9c96e5678096e6.eot?#iefix")format("embedded-opentype"),
  url("https://db.onlinewebfonts.com/t/13bf1d3e6c7119293f9c96e5678096e6.woff2")format("woff2"),
  url("https://db.onlinewebfonts.com/t/13bf1d3e6c7119293f9c96e5678096e6.woff")format("woff"),
  url("https://db.onlinewebfonts.com/t/13bf1d3e6c7119293f9c96e5678096e6.ttf")format("truetype"),
  url("https://db.onlinewebfonts.com/t/13bf1d3e6c7119293f9c96e5678096e6.svg#Uniqlo Pro Light Regular")format("svg");
}

/* 1. Use a more-intuitive box-sizing model */
*, *::before, *::after {
  box-sizing: border-box;
}

/* 2. Remove default margin */
* {
  margin: 0;
}

/* 3. Enable keyword animations */
@media (prefers-reduced-motion: no-preference) {
  html {
    interpolate-size: allow-keywords;
  }
}

body {
  /* 4. Add accessible line-height */
  line-height: 1.5;
  /* 5. Improve text rendering */
  -webkit-font-smoothing: antialiased;

  font-family: "Helvetica Neue", sans-serif;
}

/* 6. Improve media defaults */
img, picture, video, canvas, svg {
  display: block;
  max-width: 100%;
}

/* 7. Inherit fonts for form controls */
input, button, textarea, select {
  font: inherit;
}

/* 8. Avoid text overflows */
p, h1, h2, h3, h4, h5, h6 {
  overflow-wrap: break-word;
}

/* 9. Improve line wrapping */
p {
  text-wrap: pretty;
}
h1, h2, h3, h4, h5, h6 {
  text-wrap: balance;
}

/*
  10. Create a root stacking context
*/
#root, #__next {
  isolation: isolate;
}

body {
  max-width: 1200px;
  margin: 0 auto;
  padding: 0;
  text-align: center;
}

main {
  padding: 0 1rem;
}