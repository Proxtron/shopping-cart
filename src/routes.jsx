import { createBrowserRouter } from 'react-router'
import App from './App';
import HomePage from './feature/home/HomePage';
import CatalogPage from './feature/catalog/CatalogPage';
import CartPage from "./feature/cart/CartPage";
import ErrorPage from './ErrorPage';

const routes = [
  { 
    path: "/", 
    element: <App/>,
    errorElement: <ErrorPage/>,
    children: [
      { index: true, element: <HomePage/> },
      { path: "catalog", element: <CatalogPage/>},
      { path: "cart", element: <CartPage/>}
    ]
  }
];

const router = createBrowserRouter(routes);

export default router;
export {routes};