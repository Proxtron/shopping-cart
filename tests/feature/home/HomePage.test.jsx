import { describe, it, expect } from "vitest"
import { render, screen } from "@testing-library/react"
import { createMemoryRouter, RouterProvider } from "react-router"
import { routes } from "../../../src/routes"

describe("HomePage component", () => {
    it("renders correctly", () => {
        const router = createMemoryRouter(routes, {initialEntries: ["/"]});

        const {container} = render(<RouterProvider router={router}/>);
        expect(container.firstChild).toMatchSnapshot();
    });
})
