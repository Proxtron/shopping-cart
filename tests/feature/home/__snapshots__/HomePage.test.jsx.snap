// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomePage component > renders correctly 1`] = `
<header
  class="_header_c92180"
>
  <nav
    class="_headerNav_c92180"
  >
    <a
      class="_headerLink_c92180"
      data-discover="true"
      href="/"
    >
      <img
        alt="Back to home page"
        class="_headerLogo_c92180"
        src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20viewBox='0%200%2090%2040'%20width='90'%20height='40'%3e%3ctitle%3eUNIQLO%20home%3c/title%3e%3cpath%20fill='red'%20d='M50%200h40v40H50zM0%200h40v40H0z'%3e%3c/path%3e%3cg%20fill='%23fff'%3e%3cpath%20d='M79.48%205.47h2.53v12.64h-2.53zM63.47%2013.9a4.21%204.21%200%200%201-8.42%200V5.47h2.53v8.43a1.68%201.68%200%201%200%203.36%200V5.47h2.53zm11.79%2020.63h-8.42V21.89h2.53V32h5.89v2.53zm0-16.42h-2.53l-3.36-7.22v7.22h-2.53V5.47h2.53l3.36%207.22V5.47h2.53v12.64zm-16%203.78a4.21%204.21%200%200%200-4.21%204.22v4.21a4.21%204.21%200%200%200%204.21%204.21%204.34%204.34%200%200%200%20.82-.07l.86%202.6h2.53l-1.25-3.75a4.2%204.2%200%200%200%201.25-3v-4.2a4.21%204.21%200%200%200-4.21-4.22m1.68%208.43a1.68%201.68%200%201%201-3.36%200v-4.21a1.68%201.68%200%201%201%203.36%200zm19.8-8.43a4.22%204.22%200%200%200-4.22%204.22v4.21a4.21%204.21%200%200%200%208.42%200v-4.21a4.21%204.21%200%200%200-4.21-4.22m1.68%208.43a1.68%201.68%200%200%201-3.37%200v-4.21a1.68%201.68%200%200%201%203.37%200zM22.74%2015.16H34.1v2.52H22.74zM24%205.47h8.84V8H24zm-9.26%200H7.15V8h5.06v7.16H5.9v2.52h11.36v-2.52h-2.52V5.47zm8%2016.84v12.22H34.1V22.31zM31.57%2032h-6.31v-7.16h6.31zM7.15%2022.31l-1.28%206.12h2.52l.76-3.59h5.07L12.73%2032H5.14l-.51%202.53h10.11l2.52-12.22H7.15z'%3e%3c/path%3e%3c/g%3e%3c/svg%3e"
      />
    </a>
    <a
      class="_headerLink_c92180 
                    false
                "
      data-discover="true"
      href="/catalog"
    >
      Catalog
    </a>
    <a
      class="_headerLink_c92180
                    false
                "
      data-discover="true"
      href="/cart"
    >
      <img
        alt="Navigate to Cart Page"
        src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20width='24'%20height='24'%20fill='none'%20viewBox='0%200%2024%2024'%20class='ito-primary-icon-color'%3e%3cpath%20fill='currentColor'%20fill-rule='evenodd'%20d='m7.044%206.566-.004-3.06H2.008v1.199h3.834l.014%2010.754h12.897l2.24-8.894zm10.774%207.694H7.055l-.008-6.495h12.407zm-.85%206.242a1.5%201.5%200%201%200%200-3%201.5%201.5%200%200%200%200%203m-8.015-1.5a1.5%201.5%200%201%201-3%200%201.5%201.5%200%200%201%203%200'%20clip-rule='evenodd'%3e%3c/path%3e%3c/svg%3e"
      />
    </a>
  </nav>
</header>
`;
